src/python/pants/backend/javascript/nodejs_project_environment.py

```diff
@@ -54,7 +54,9 @@ def root_dir(self) -> str:
     def node_modules_directories(self) -> Iterable[str]:
         yield "node_modules"
         if self.package and not self.project.single_workspace:
-            yield os.path.join(self.relative_workspace_directory(), "node_modules")
+            yield os.path.join(
+                os.path.relpath(os.path.curdir, self.relative_workspace_directory()), "node_modules"
+            )
```

Comment:
So this was very broken before. The `NodeJSProjectEnvironment` assumes that
1. CWD should always be at the project root dir, but
2. that all output files and output directories will be referred to relative to the _workspace_

2. Is true for "goal facing" stuff like lockfile generation, package goals and now tests. It is very convenient there. But I had borked it here, and made it so this method produce paths relative to 1 instead of 2.

Reply:
Hm. Yea, it's slightly odd to try and have this both ways.

Do JS runners have a concept like [source roots](https://www.pantsbuild.org/docs/source-roots) (aka the PYTHONPATH, aka the CLASSPATH), where a prefix is stripped from a directory? If so, it might be good to make lockfile generation and test running consistently use strategy (2) using that facility.

That would also ease eventually allowing source dependencies between JS projects in a monorepo, IIUC.

Reply:
Ok, it sounds like unlike in Python (and similar to the JVM), there is an install step for dependency projects that means that they can't collide with source dependencies.